Mathlib/CategoryTheory/Limits/Shapes/FunctorToTypes.lean

```diff
@@ -0,0 +1,261 @@
+/-
+Copyright (c) 2024 Jack McKoen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jack McKoen
+-/
+import Mathlib.CategoryTheory.Limits.FunctorCategory
+import Mathlib.CategoryTheory.Limits.Types
+
+/-!
+# Binary (co)products of type-valued functors
+
+Defines an explicit construction of binary products and coproducts of type-valued functors.
+
+Also defines isomorphisms to the categorical product and coproduct, respectively.
+-/
+
+
+open CategoryTheory.Limits
+
+universe w v u
+
+namespace CategoryTheory.FunctorToTypes
+
+variable {C : Type u} [Category.{v} C]
+
+section prod
+
+/-- `prod F G` is the explicit binary product of type-valued functors `F` and `G`. -/
+def prod (F G : C ⥤ Type w) : C ⥤ Type w where
```

Comment:
As `(F G : C ⥤ Type w)` are variables throughout most of the file, it would be slightly better to add a line like this by the top of the file:
```lean
variable (F G : C ⥤ Type w)
```
Then, you may replace `def prod (F G : C ⥤ Type w) : C ⥤ Type w where` by `def prod : C ⥤ Type w where`. When you need to make parameters implicit or explicit, you may do `variable {F G}` or `variable (F G)`.

Reply:
Okay I added `F` and `G` as variables. Through the rest of the file I just alternated writing `variable {F G}` and `variable (F G)` whenever necessary (about 8 times), which I'm not sure is good practice. This seems a little excessive, so maybe I misunderstood what you meant.

Reply:
This seems good to me. (When only one definition needs such change, we could also do `variable {F G} in def ....`, but it is not necessary here.)